### DISCUSSION TITLE
Tensor Database mangagement systems

### DISCUSSION THREAD

**Discussion by ghost** [community/community#122832] [Programming Help] [answered]
### Select Topic Area

Question

### Body

i am looking for a software that main job is tensor manipulations and act as DBMS

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @AElnamaki ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by CopperEagle** [comment, marked as answer]
Hey @AElnamaki 

If you use Python, you can use SQLite3, which is in the standard library. You can most of the familiar SQL language but you can store any kind of Python objects, including Pytorch.tensor, or any other type of tensor objects.

To quote the documentation of sqlite3 itself, added with comments for viewing

```py
# To use the module, start by creating a database Connection object:
        
import sqlite3
cx = sqlite3.connect("test.db")  # test.db will be created or opened
    
# The special path name ":memory:" can be provided to connect to a transient
#  in-memory database:
    
# cx = sqlite3.connect(":memory:")  # connect to a database in RAM
    
# Once a connection has been established, create a Cursor object and call
# its execute() method to perform SQL queries:
    
cu = cx.cursor()
    
# create a table
cu.execute("create table lang(name, first_appeared)")
    
# insert values into a table
# [INSERT TENSOR HERE]
cu.execute("insert into lang values (?, ?)", ("C", 1972))
    
# execute a query and iterate over the result
for row in cu.execute("select * from lang"):
    print(row)
    
cx.close()
```

Hope this answers your question. If it did, please consider marking it as the answer.

**Reply by CopperEagle** [reply]
@AElnamaki I guess you selected the wrong answer :laughing:  ? Anyways, glad to help :smile:
